src/components/Sender.tsx

```diff
@@ -30,78 +33,98 @@ interface Props {
   className?: string;
 }
 
-const Accounts = ({ className }: Props) => {
-  const [chains, setChains] = useState<Array<string>>([]);
+const useStyles = makeStyles(() => ({
+  row: {
+    minWidth: '100%',
+    margin: '10px 0'
+  }
+}));
+
+const Sender = ({ className }: Props) => {
+  const classes = useStyles();
   const { account, accounts, derivedAccount, setCurrentAccount } = useAccounts();
   const {
     sourceChainDetails: { sourceChain },
     targetChainDetails: { targetChain }
   } = useSourceTarget();
-
-  useEffect(() => {
-    if (!chains.length) {
-      setChains([sourceChain, targetChain]);
-    }
-  }, [chains.length, sourceChain, targetChain]);
+  const { setReceiver } = useReceiver();
 
   const value = account ? encodeAddress(account.address, getChainSS58(sourceChain)) : '';
 
   const onChange = (value: string, chain: string) => {
     setCurrentAccount(value, chain);
+    setReceiver(null);
   };
-  const renderAccounts = (chain: string) => {
-    const formatedAccounts = formatAccounts(accounts, chain);
+
+  const renderAccounts = (source: string, target: string) => {
+    const formatedAccounts = formatAccounts(accounts, source);
     const items = formatedAccounts.map(({ text, value, key }: any) => (
       <MenuItem
         key={key}
         value={value}
         onClick={() => {
-          onChange(value, chain);
+          onChange(value, source);
         }}
       >
-        <Account text={text} value={value} showDerivedBalance chain={chain} />
+        <div className={classes.row}>
+          <Account accountName={text} value={value} chain={source} />
+          <Account accountName={text} value={value} isDerived fromSender chain={target} />
```

Comment:
`fromSender` just displays balance, why bother with name and `isDerived`?

Reply:
Completely removed.